vsintegration/tests/FSharp.Editor.Tests/SemanticClassificationServiceTests.fs

```diff
@@ -183,3 +187,56 @@ let g() =
 """
 
         verifyNoClassificationDataAtEndOfMarker (sourceText, marker, classificationType)
+
+    [<Theory>]
+    [<InlineData("(*1*)", ClassificationTypeNames.Keyword)>]
+    [<InlineData("(*2*)", ClassificationTypeNames.Keyword)>]
+    [<InlineData("(*3*)", ClassificationTypeNames.Keyword)>]
+    [<InlineData("(*4*)", ClassificationTypeNames.LocalName)>]
+    [<InlineData("(*5*)", ClassificationTypeNames.LocalName)>]
+    [<InlineData("(*6*)", ClassificationTypeNames.LocalName)>]
+    [<InlineData("(*7*)", ClassificationTypeNames.Identifier)>]
+    [<InlineData("(*8*)", ClassificationTypeNames.Identifier)>]
+    [<InlineData("(*9*)", ClassificationTypeNames.ClassName)>]
+    [<InlineData("(*10*)", ClassificationTypeNames.ClassName)>]
+    [<InlineData("(*11*)", ClassificationTypeNames.ClassName)>]
+    [<InlineData("(*12*)", ClassificationTypeNames.ClassName)>]
+    [<InlineData("(*13*)", ClassificationTypeNames.ClassName)>]
+    [<InlineData("(*14*)", ClassificationTypeNames.TypeParameterName)>]
+    [<InlineData("(*15*)", ClassificationTypeNames.TypeParameterName)>]
+    [<InlineData("(*16*)", ClassificationTypeNames.Keyword)>]
+    [<InlineData("(*17*)", ClassificationTypeNames.Keyword)>]
+    [<InlineData("(*18*)", ClassificationTypeNames.Keyword)>]
+    member _.``nameof ident, nameof<'T>, match … with nameof ident``(marker: string, classificationType: string) =
+        let sourceText =
+            """
+module ``Normal usage of nameof should show up as a keyword`` =
+    let f x = (*1*)nameof x
+    let g (x : 'T) = (*2*)nameof<'T>
+    let h x y = match x with (*3*)nameof y -> () | _ -> ()
+
+module ``Redefined nameof should shadow the intrinsic one`` =
+    let a x = match x with (*4*)nameof -> ()
+    let b (*5*)nameof = (*6*)nameof
+    let (*7*)nameof = "redefined"
+    let _ = (*8*)nameof
+
+    type (*9*)nameof () = class end
+    let _ = (*10*)nameof ()
+    let _ = new (*11*)nameof ()
+
+    module (*12*)nameof =
+        let f x = x
+
+    let _ = (*13*)nameof.f 3
+
+    let c (x : '(*14*)nameof) = x
+    let d (x : (*15*)'nameof) = x
+
+module ``It should still show up as a keyword even if the type parameter is invalid`` =
+    let _ = (*16*)nameof<>
```

Comment:
Why is it colored as a keyword? It's a function in F#.

Reply:
Yeah, FCS classifies `nameof` as `SemanticClassificationType.IntrinsicFunction` (assuming the name resolution has been recorded properly):

https://github.com/dotnet/fsharp/blob/ad29712d1ac6f026711cb585e0d184a190890e4c/src/Compiler/Service/SemanticClassification.fs#L34

https://github.com/dotnet/fsharp/blob/ad29712d1ac6f026711cb585e0d184a190890e4c/src/Compiler/Service/SemanticClassification.fs#L157-L168

https://github.com/dotnet/fsharp/blob/ad29712d1ac6f026711cb585e0d184a190890e4c/src/Compiler/Service/SemanticClassification.fs#L218

But VS maps that to its own, separate classification, in the same category as keywords:

https://github.com/dotnet/fsharp/blob/ad29712d1ac6f026711cb585e0d184a190890e4c/vsintegration/src/FSharp.Editor/Classification/ClassificationDefinitions.fs#L57